lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -205,135 +210,223 @@ def _on_list_backups_action(self, event: ActionEvent) -> None:
         else:
             event.fail("Failed: invalid output format, must be either json or table")
 
-    def _restore_and_try_close_indices_if_needed(
-        self, backup_id: int
-    ) -> Tuple[Dict[str, Any], Set[str]]:
-        """Restores and processes any exception related to running indices.
+    def _request_indices_to_close(self, indices: Set[str]) -> Dict[str, Any]:
+        """Requests indices to be closed and return their status."""
+        resp = self._request(
+            "POST",
+            f"{','.join(indices)}/_close",
+            payload={
+                "ignore_unavailable": "true",
+            },
+        )
+        if not resp or not resp.get("acknowledged", False):
+            return {"acknowledged": False}
+        closed_indices_state = {
+            index: IndexStateEnum.CLOSED
+            if state and state.get("closed") == "true"
+            else IndexStateEnum.OPEN
+            for index, state in resp.get("indices", {}).items()
+        }
+        return {"acknowledged": True, "indices": closed_indices_state}
+
+    def _close_indices_if_needed(self, backup_id: int) -> Set[str]:
+        """Closes indices that will be restored.
```

Comment:
I agree these methods need a renaming. But  I prefer to keep this method as its actual go: closes indices if it is really needed; and we have a method that actually does the API call.

Otherwise I need to bring the close Retry logic up to the restore action, which is already big enough.